core/services/runs.go

```diff
@@ -35,13 +38,13 @@ func markInProgressIfSufficientIncomingConfirmations(run *models.JobRun, taskRun
 	}
 }
 
-func validateOnMainChain(run *models.JobRun, taskRun *models.TaskRun, txManager store.TxManager) error {
+func validateOnMainChain(run *models.JobRun, taskRun *models.TaskRun, ethClient eth.Client) error {
 	txhash := run.RunRequest.TxHash
 	if txhash == nil || !taskRun.MinRequiredIncomingConfirmations.Valid || taskRun.MinRequiredIncomingConfirmations.Uint32 == 0 {
 		return nil
 	}
 
-	receipt, err := txManager.GetTxReceipt(*txhash)
+	receipt, err := ethClient.TransactionReceipt(context.TODO(), *txhash)
```

Comment:
Did this for two reasons: 1) reduce scope of this PR while signalling that we need to address this, and 2) because we already have a story up on the board to introduce contexts everywhere.  Seemed to fit better under that heading.